hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/IngestChecker.java

```diff
@@ -104,21 +123,83 @@ public TransactionInfo runAllChecks(@NonNull final HederaState state, @NonNull f
 
         // 3. Check throttles
-        if (throttleAccumulator.shouldThrottle(transactionInfo.txBody())) {
-            throw new PreCheckException(ResponseCodeEnum.BUSY);
+        if (throttleAccumulator.shouldThrottle(txInfo.txBody())) {
+            throw new PreCheckException(BUSY);
         }
 
         // 4. Get payer account
-        final AccountID payerID =
+        final AccountID payerId =
                 txBody.transactionIDOrElse(TransactionID.DEFAULT).accountIDOrElse(AccountID.DEFAULT);
-        solvencyPreCheck.checkPayerAccountStatus(state, payerID);
+
+        solvencyPreCheck.checkPayerAccountStatus(state, payerId);
 
         // 5. Check account balance
         solvencyPreCheck.checkSolvencyOfVerifiedPayer(state, tx);
 
         // 6. Verify payer's signatures
-        signaturePreparer.syncGetPayerSigStatus(tx);
+        verifyPayerSignature(state, txInfo, payerId);
+
+        return txInfo;
+    }
+
+    private void verifyPayerSignature(
+            @NonNull final HederaState state, @NonNull final TransactionInfo txInfo, @NonNull final AccountID payerId)
+            throws PreCheckException {
+
+        // Get the payer account
+        final var stores = new ReadableStoreFactory(state);
+        final var payerAccount = stores.getStore(ReadableAccountStore.class).getAccountById(payerId);
+
+        // If there is no payer account for this ID, then the transaction is invalid
+        if (payerAccount == null) {
+            throw new PreCheckException(INVALID_ACCOUNT_ID);
+        }
+
+        // There should, absolutely, be a key for this account. If there isn't, then something is wrong in
+        // state. So we will log this with a warning. We will also have to do something about the fact that
+        // the key is missing -- so we will fail with unauthorized.
+        if (!payerAccount.hasKey()) {
+            // FUTURE: Have an alert and metric in our monitoring tools to make sure we are aware if this happens
+            logger.warn("Payer account {} has no key, indicating a problem with state", payerId);
+            throw new PreCheckException(UNAUTHORIZED);
+        }
+
+        // Expand the signatures
+        final var expandedSigs = new HashSet<ExpandedSignaturePair>();
+        signatureExpander.expand(
+                payerAccount.keyOrThrow(), txInfo.signatureMap().sigPairOrThrow(), expandedSigs);
 
-        return transactionInfo;
+        // Verify the signatures
+        final var results = signatureVerifier.verify(txInfo.signedBytes(), expandedSigs);
+        final var future = results.get(payerAccount.keyOrThrow());
+
+        // This can happen if the signature map was missing a signature for the payer account.
+        if (future == null) {
+            throw new PreCheckException(INVALID_SIGNATURE);
+        }
+
+        // Wait for the verification to complete. We have a timeout here of 1 second, which is WAY more time
```

Comment:
This is done in a background thread. Does it matter if we wait 1 or 10 seconds? GC pauses are rare but happen.